helix-core/src/indent.rs

```diff
@@ -337,6 +344,35 @@ fn query_indents(
     let mut indent_captures: HashMap<usize, Vec<IndentCapture>> = HashMap::new();
     let mut extend_captures: HashMap<usize, Vec<ExtendCapture>> = HashMap::new();
     cursor.set_byte_range(range);
+
+    enum End {
+        Start,
```

Comment:
Not a huge fan of nested items like this. Can this just be a normal enum and a normal function instead of a closure? I only noticed one capture that could easily br another argument.

I personally also think that an enum makes the whole thing a bit confusing. The function is short enough where having two functions might be more readable.